### PR TITLE
Android design library v2.0.0 based on Material

### PR DESCRIPTION
**Pull request by mani-w3w**
Breaking changes since we are moving from a proprietary design system to Material Design System.

Works with 2 themes MaterialTheme and W3WTheme (which is based on material with different colorSchemes). 

Sample app to represent what we have currently on Figma, and that allows you to swap between themes and day/night. 

These are the tasks covered in this PR didn't separate them due to difficult segmentation since it is such a breaking change:

- https://what3words.atlassian.net/browse/MT-6251
- https://what3words.atlassian.net/browse/MT-6252
- https://what3words.atlassian.net/browse/MT-6246
- https://what3words.atlassian.net/browse/MT-6237

This is not yet finished completely. We are still waiting on some changes from the design team, but we want to get feedback from the team for now and will create a fix/ PR if some changes come through next week.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,17 +1,23 @@
 buildscript {
     ext {
-        compose_version = '1.4.3'
+        compose_version = '1.6.0-beta03'
+        material3_version = "1.2.0-beta01"
```
Comment (phpduy) [inline build.gradle:4]: The compose version 1.6.0 has been officially released. I think we can start using it.
There is also newer material3_version:1.2.0-rc01

**Review by phpduy — CHANGES_REQUESTED** [review]

File: design-library-usage-sample/src/main/java/com/what3words/design/library/sample/ui/screens/IconButtonScreen.kt [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+ * A screen showcasing various styles of icon buttons in Material Design styled by both [MaterialTheme] and [W3WTheme].
+ * This screen displays different types of icon buttons including standard, filled,
+ * tonal, and outlined. Each button is presented in both its enabled and disabled states
+ * to demonstrate the visual changes in different states.
+ *
+ * This screen is useful for designers and developers to understand the styling options
+ * and states of icon buttons in a Material Design-based UI.
+ */
+@Composable
+fun IconButtonScreen() {
```
Comment (prosper-w3w) [inline design-library-usage-sample/src/main/java/com/what3words/design/library/sample/ui/screens/IconButtonScreen.kt:35]: In adherence to Jetpack Compose API guidelines, it is advised that every composable function should embrace and adhere to the inclusion of a Modifier-type parameter.

The guidelines state, "_Element functions MUST accept a parameter of type Modifier. This parameter MUST be named “modifier” and MUST appear as the first optional parameter in the element function's parameter list. Element functions MUST NOT accept multiple Modifier parameters._" Further details can be found [here](https://android.googlesource.com/platform/frameworks/support/+/androidx-main/compose/docs/compose-api-guidelines.md#elements-accept-and-respect-a-modifier-parameter:~:text=Elements%20accept%20and,multiple%20Modifier%20parameters.)

File: build.gradle (hunk 1 above)
Comment (mani-w3w) [reply build.gradle]: Done 👍

File: design-library-usage-sample/src/main/java/com/what3words/design/library/sample/ui/screens/ColorPaletteScreen.kt [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+ * This composable function displays various color combinations from the MaterialTheme color scheme,
+ * including primary, secondary, error, warning, success, surface, and background colors.
+ * Each color is demonstrated with its corresponding on-color (text color) for contrast.
+ *
+ * This screen serves as a visual guide to understanding the color scheme and how colors interact
+ * with their respective on-colors, which is critical for maintaining accessibility and readability
+ * in UI design.
+ */
+@Composable
+fun ColorPaletteScreen() {
```
[the remark this answers is not in the document]
Comment (mani-w3w) [reply design-library-usage-sample/src/main/java/com/what3words/design/library/sample/ui/screens/ColorPaletteScreen.kt]: Done, good point mate, added to all screens 👍

**Review by kaytran2992 — APPROVED** [review]
LGTM

**Pull request merged**